src/ProtocolFees.sol

```diff
@@ -78,18 +78,13 @@ abstract contract ProtocolFees is IProtocolFees, Owned {
             bytes memory data = abi.encodeCall(IProtocolFeeController.protocolFeeForPool, (key));
             uint256 returnData;
             assembly ("memory-safe") {
-                success := call(gasLimit, toAddress, 0, add(data, 0x20), mload(data), 0, 0)
+                // only load the first 32 bytes of the return data to prevent gas griefing
+                success := call(gasLimit, toAddress, 0, add(data, 0x20), mload(data), 0, 32)
```

Comment:
awesome!

Reply:
```suggestion
                success := call(gasLimit, toAddress, 0, add(data, 0x20), mload(data), 0, 0x20)
```
nit: use hex for calldata/memory offsets for consistency.